intelmq/bots/parsers/turris/parser.py

```diff
@@ -1,50 +1,49 @@
-# SPDX-FileCopyrightText: 2015 robcza
+# SPDX-FileCopyrightText: 2023 Filip Pokorný
 #
 # SPDX-License-Identifier: AGPL-3.0-or-later
 
-# -*- coding: utf-8 -*-
-import csv
-import io
-
-from intelmq.lib import utils
 from intelmq.lib.bot import ParserBot
 
+PORTS = {
+    "ftp": 21,
+    "telnet": 23,
+    "http": 80
+    # smtp uses both 25 and 587, therefore we can't say for certain
+}
+
 
 class TurrisGreylistParserBot(ParserBot):
     """Parse the Turris Greylist feed"""
 
-    def process(self):
-        report = self.receive_message()
-
-        columns = [
-            "source.ip",
-            "source.geolocation.cc",
-            "event_description.text",
-            "source.asn"
-        ]
-
-        headers = True
-        raw_report = utils.base64_decode(report.get("raw"))
-        raw_report = raw_report.translate({0: None})
-        for row in csv.reader(io.StringIO(raw_report)):
-            # ignore headers
-            if headers:
-                headers = False
-                continue
+    parse = ParserBot.parse_csv
+    recover_line = ParserBot.recover_line_csv
+    _ignore_lines_starting = ["#", "Address"]
```

Comment:
is the line starting with `Address` the header? If yes, you may also use `parse_csv_dict` which parses the header and uses it as field names for the rows

Reply:
Thanks! Updated.